### ISSUE TITLE
Tag links to wrong location

### ISSUE THREAD

**Issue by magikid** [writeas/writefreely#335]
### Describe the bug
I run a single user instance.  When I add the `#infra` hashtag to a post, it links to `/tag:infra` which throws a 404 not found.  Here's the post that I noticed it on: https://www.christopherjones.us/please-excuse-the-mess

### Steps to reproduce (if necessary)
1. Create a new post
2. Add a hashtag
3. Publish the post
4. Click on the hashtag link and see that you get a 404.

### Expected behavior
I expected to be shown a list of tagged posts.

### Application configuration
 - **Single mode or Multi-user mode?** Single mode
 - **Database?** mysql
 - **Open registration?** No
 - **Federation enabled?** Yes

**Version or last commit**: 
```
$ /opt/writefreely/writefreely --version
WriteFreely 0.12.0
```

<details>
<summary>Here's a copy of my config too</summary>

```
[server]
hidden_host          = 
port                 = 47635
bind                 = 0.0.0.0
tls_cert_path        = 
tls_key_path         = 
autocert             = false
templates_parent_dir = 
static_parent_dir    = 
pages_parent_dir     = 
keys_parent_dir      = 
hash_seed            = 

[database]
type     = mysql
filename = 
username = name
password = password
database = dbname
host     = host
port     = port

[app]
site_name          = Chris's Thoughts
site_description   = 
host               = https://www.christopherjones.us
theme              = write
editor             = 
disable_js         = false
webfonts           = true
landing            = 
simple_nav         = false
wf_modesty         = true
chorus             = false
forest             = false
disable_drafts     = false
single_user        = true
open_registration  = false
min_username_len   = 3
max_blogs          = 1
federation         = true
public_stats       = true
private            = false
local_timeline     = false
user_invites       = 
default_visibility = 
update_checks      = true

[oauth.slack]
client_id          = 
client_secret      = 
team_id            = 
callback_proxy     = 
callback_proxy_api = 

[oauth.writeas]
client_id          = 
client_secret      = 
auth_location      = 
token_location     = 
inspect_location   = 
callback_proxy     = 
callback_proxy_api = 

[oauth.gitlab]
client_id          = 
client_secret      = 
host               = 
display_name       = 
callback_proxy     = 
callback_proxy_api = 
```
</details>

**Comment by magikid** [comment]
I turned on debug and hit the tag location and found this in the logs:
```
Jul 14 10:51:02 webby.chriswjon.es writefreely[6523]: ERROR: 2020/07/14 10:51:02 database.go:1223: Error after Next() on rows: Error 3685: Illegal argument to a regular expression.
Jul 14 10:51:02 webby.chriswjon.es writefreely[6523]: 2020/07/14 10:51:02 "GET /tag:infra" 404 206.807204ms "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.116 Safari/537.36"
```

**Comment by thebaer** [comment]
Thanks for the detailed report, @magikid! Which version of MySQL are you running? My first instinct is that something has changed there, so our regex in this query no longer works: https://github.com/writeas/writefreely/blob/99d86a74894586e574a86d656940d58d9dd50112/database.go#L1199

**Comment by magikid** [comment]
Ah, good call! I'm running MySQL 8.

When I run the query on the server, it gives that same error:
```
SELECT *
FROM posts
WHERE collection_id = 1
    AND LOWER(content) RLIKE '#infra[[:>:]]'
    ORDER BY created desc
```

Returns 
```
Illegal argument to a regular expression.

Error code 3685.
```

**Comment by magikid** [comment]
I found the answer in the [MySQL docs for 8](https://dev.mysql.com/doc/refman/8.0/en/regexp.html#regexp-compatibility):

>The Spencer library [used by 5.6] supports word-beginning and word-end boundary markers ([[:<:]] and [[:>:]] notation). ICU [used by 8] does not. For ICU, you can use \b to match word boundaries; double the backslash because MySQL interprets it as the escape character within strings.

**Comment by magikid** [comment]
@thebaer Is this something that you'd accept a patch for?

**Comment by thebaer** [comment]
Thanks for the research! Yes, we'd absolutely accept a patch for this. I'd just say that ideally, it should maintain compatibility back to MySQL 5.6.

**Comment by wiwie** [comment]
So a workaround for now is to run writefreely with MySQL < 8?

**Comment by magikid** [comment]
@wiwie That was my resolution. I switched from MySQL 8 to MySQL 5.6 and my writefreely instance started working just fine.

On a side note, I’ll try to get a PR sent over to update the docs about this issue but I’m not terribly proficient at golang so I won’t be submitting a patch to actually fix this.

**Comment by wiwie** [comment]
@magikid you could migrate all your data from 8 to 5.6 without problems? Then I'll give that a try as well.

**Comment by magikid** [comment]
@wiwie Yes, I was able to switch to the older version pretty easily but I had basically nothing in the database yet. 

And though I remember doing the migration while I was testing this, it seems that I’d forgotten how I actually fixed this. I just logged into the server and it seems that the way I actually fixed this was to clone the project, change the sql query and then recompile. 

I’ll try to post a patch later today.

**Comment by DearRude** [comment]
I also have this issue on SQLite with version 0.12
Any updates on this? @thebaer

**Comment by thebaer** [comment]
Similar to #219, we'll likely have this permanently fixed with a new upcoming tag storage system: [T809](https://phabricator.write.as/T809).

**Comment by Poslovitch** [comment]
Hey there, looks like this is running on Write.as for a while. When is this fix going to be available to broader use?

**Comment by PauKokura** [comment]
Sorry to say, this is still a bug in v.0.15.0.

**Comment by thebaer** [comment]
Closing, as this was fixed in #1461 / v0.16.0.
